dnsext-iterative/DNS/Iterative/Server/Pipeline.hs

```diff
@@ -311,24 +314,25 @@ type VcRespAvail = STM Bool
 type VcAllowInput = STM Bool
 
 {- FOURMOLU_DISABLE -}
-data VcTimeout =
-    VcTimeout
+data VcTimer =
+    VcTimer
     { vtManager_        :: TimerManager
     , vtKey_            :: TimeoutKey
-    , vtState_          :: TVar Bool
     , vtMicrosec_       :: Int
     }
 {- FOURMOLU_ENABLE -}
 
 {- FOURMOLU_DISABLE -}
+-- separate the STM computations that determine event states from
+-- the conditions and parameters that cause these events to occur.
+-- place only what is directly necessary for STM calculations in `VcSession`.
 data VcSession =
     VcSession
     { vcEof_            :: VcEof
     -- ^ EOF is received. This is an RX evet.
-    , vcTimeout_        :: VcTimeout
+    , vcTimeout_        :: TVar Bool
```

Comment:
Please give a name to this field in the same way as the other fields.
Maybe `VcTimedOut`?

Reply:
I believe that type synonyms that are simply named the same as their terms are not a very good idea, as they provide little information from the type.
For example, how about updating it to something like `TVar VcEof` and `TVar VcTimeout`?

Reply:
I don't understand your point.
What I'm saying is "do as other fields do".
`type VcEof = TVar Bool` is used for `vcEof_`.
So, `type Something = TVar Bool` should be used for `vcTimeout_`.

Reply:
I believe it's more understandable to have types that show TVar and STM.

https://github.com/kazu-yamamoto/dnsext/commit/0742e2d991105d96b4a142732abb8c3d4f26bff3

Reply:
https://github.com/kazu-yamamoto/dnsext/commit/0742e2d991105d96b4a142732abb8c3d4f26bff3 is OK for me since all fields are in the same style.